src/modules/site-v2/base/utils/tools.py

```diff
@@ -58,63 +63,77 @@ def lookup_report(kind, reportId, user=None, validate_user=True):
       raise ReportLookupError('You do not have access to that report.', 401)
 
   # If the user doesn't have permission to view this report, show an error message
-  if validate_user and not (report.username == user.name or user_is_admin()):
+  if validate_user and not (job.report.username == user.name or user_is_admin()):
     raise ReportLookupError('You do not have access to that report.', 401)
 
   # If all checks passed, return the report entity
-  return report
+  return job
 
 
-def try_submit(EntityClass, user, data, no_cache):
+def try_submit(kind, user, data, no_cache):
 
-  # Try submitting the job
   try:
-    report = submit_job(EntityClass, user, data, no_cache=no_cache, valid_file_extensions=TOOL_INPUT_DATA_VALID_FILE_EXTENSIONS)
+    # Try creating the job
+    #   DuplicateDataError -> this user has already submitted this job
+    #   DataFormatError    -> there is an error in the input data
+    job = get_pipeline_class(kind=kind).create(user, data, no_cache=no_cache, valid_file_extensions=TOOL_INPUT_DATA_VALID_FILE_EXTENSIONS)
+
+    # Schedule the job, if applicable
+    #   CachedDataError    -> the job already has results
+    if job.is_schedulable:
+      job.schedule()
+      ready = False
+
+    # If job is not schedulable, its results are immediately available, i.e. ready now
+    else:
+      ready = True
+
     return {
       'cached':    False,
       'same_user': None,
-      'ready':     False,
-      'data_hash': report.data_hash,
-      'id':        report.id,
+      'ready':     ready,
+      'data_hash': job.report.data_hash,
+      'id':        job.report.id,
     }, 200
 
   # Duplicate job submission from this user
   except DuplicateDataError as ex:
+    job = ex.handler
 
     # Log the event
-    logger.debug(f'(CACHE HIT) User resubmitted duplicate {EntityClass.kind} data: id = {ex.report.id}, data hash = {ex.report.data_hash}, status = {ex.report["status"]}')
+    logger.debug(f'(CACHE HIT) User resubmitted duplicate {kind} data: id = {job.report.id}, data hash = {job.report.data_hash}, status = {job.report["status"]}')
 
     # Return the matching entity
     return {
       'cached':    True,
       'same_user': True,
-      'ready':     ex.report.is_finished(),
-      'data_hash': ex.report.data_hash,
-      'id':        ex.report.id,
+      'ready':     job.report.is_finished(),
+      'data_hash': job.report.data_hash,
+      'id':        job.report.id,
       'message':   'You have already submitted this data file. Here\'s your previously generated report.',
     }, 200
 
   # Duplicate job submission from another user
   except CachedDataError as ex:
```

Comment:
I'll admit this is slightly hack-y, but this error can only be thrown if the `job` object was instantiated correctly, so we have access to it in this branch (but not in others).  There might be a better way to rewrite this whole process without using errors, but I've left it as-is for now since it does work.